### PR TITLE
feat: docs, tests and fixes for create/update/deleteRecord builders

### PR DESCRIPTION
**Pull request by Baltazore**
## Documentation for rest of the builder functions

This PR aims to cover create-record / update-record / delete-record documentation

Should be added to json-api / rest / active-record packages

### CODE REVIEW

File: packages/json-api/src/-private/builders/save-record.ts [hunk 1]
```diff
@@ -52,6 +52,48 @@ export function deleteRecord(record: unknown, options: ConstrainedRequestOptions
...
+/**
+ * Builds request options to create new record for resources,
+ * configured for the url, method and header expectations of most JSON:API APIs.
+ *
+ * **Basic Usage**
+ *
+ * ```ts
+ * import { createRecord } from '@ember-data/json-api/request';
+ *
+ * const person = this.store.createRecord('person', { name: 'Ted' });
```
Comment (Baltazore) [inline packages/json-api/src/-private/builders/save-record.ts:64]: @runspired am I correct here that this is how new record will be created? 

It seems like I am missing something here... kinda feels not right with `this.store.createRecord` and `create-record` helper

Comment (Baltazore) [reply packages/json-api/src/-private/builders/save-record.ts:106]: confirmed on ember-data meeting

File: packages/json-api/src/-private/builders/save-record.ts [hunk 2]
```diff
@@ -52,10 +94,53 @@ export function deleteRecord(record: unknown, options: ConstrainedRequestOptions
...
+ * @for @ember-data/json-api/request
+ * @param record
+ * @param options
+ */
 export function createRecord(record: unknown, options: ConstrainedRequestOptions = {}): CreateRequestOptions {
   const identifier = recordIdentifierFor(record);
   assert(`Expected to be given a record instance`, identifier);
-  assert(`Cannot delete a record that does not have an associated type and id.`, isExisting(identifier));
+  // TODO: seems to be wrong assert here, at least message is confusing
+  // assert(`Cannot delete a record that does not have an associated type and id.`, isExisting(identifier));
```
Comment (leepfrog) [inline packages/json-api/src/-private/builders/save-record.ts:143]: +1 to wrong assertion, you should be able to create a record without an identifier.

Comment (runspired) [reply packages/json-api/src/-private/builders/save-record.ts:143]: @leepfrog not exactly. You do need an identifier, the identifier does not need an `id`.

Comment (leepfrog) [reply packages/json-api/src/-private/builders/save-record.ts:143]: Whoops!  Yes, agree!

File: tests/builders/tests/test-helper.js [hunk 3]
```diff
@@ -6,13 +8,18 @@ import assertAllDeprecations from '@ember-data/unpublished-test-infra/test-suppo
...
+import config from '../config/environment';
+
 // Handle testing feature flags
 if (QUnit.urlParams.enableoptionalfeatures) {
   window.EmberDataENV = { ENABLE_OPTIONAL_FEATURES: true };
 }
 
 configureAsserts();
 
+setApplication(Application.create(config.APP));
```
Comment (runspired) [inline tests/builders/tests/test-helper.js:21]: in follow up we should eliminate this and the new app files in favor of in-test-file registration

**Review by runspired — APPROVED** [review]

**Pull request merged**